File: engine/engine-model/src/main/scala/tools/packet/family/SeniorMessage.scala
package tools.packet.family

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class SeniorMessage(private var _characterName: String) extends PacketInput {
  def characterName: String = _characterName

  override def opcode(): SendOpcode = SendOpcode.FAMILY_JOIN_ACCEPTED
}

File: engine/engine-model/src/main/java/tools/packet/family/SeniorMessage.java
package tools.packet.family;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record SeniorMessage(String characterName) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.FAMILY_JOIN_ACCEPTED;
   }
}